anaDACScan.py

```diff
@@ -0,0 +1,311 @@
+        calibrated_ADC_value /= 1000.0
+        calibrated_ADC_error /= 1000.0
+        
+        #Use Ohm's law to convert the currents to voltages. The VFAT3 team told us that a 20k ohm resistor was used.
+        if nominalDacValues[nameX][1][1] == "A":
+
+            calibrated_ADC_value = calibrated_ADC_value/20000.0
+            calibrated_ADC_error = calibrated_ADC_error/20000.0
+
+            if nameX != 'CFG_IREF':
+                calibrated_ADC_value -= nominal_iref 
+            
+        #the reversal of x and y is intended - we want to plot the nameX variable on the y-axis and the nameY variable on the x-axis
+        #the nameX variable is the DAC register that is scanned, and we want to determine its nominal value
+        if args.assignXErrors:
+            dict_DACvsADC_Graphs[oh][vfat].SetPoint(dict_DACvsADC_Graphs[oh][vfat].GetN(),calibrated_ADC_value,event.dacValX)
+            dict_RawADCvsDAC_Graphs[oh][vfat].SetPoint(dict_RawADCvsDAC_Graphs[oh][vfat].GetN(),event.dacValY,event.dacValX)
+            dict_DACvsADC_Graphs[oh][vfat].SetPointError(dict_DACvsADC_Graphs[oh][vfat].GetN()-1,calibrated_ADC_error,event.dacValX_Err)
+            dict_RawADCvsDAC_Graphs[oh][vfat].SetPointError(dict_RawADCvsDAC_Graphs[oh][vfat].GetN()-1,event.dacValY_Err,event.dacValX_Err)
+        else:
+            dict_DACvsADC_Graphs[oh][vfat].SetPoint(dict_DACvsADC_Graphs[oh][vfat].GetN(),calibrated_ADC_value,event.dacValX)
+            dict_RawADCvsDAC_Graphs[oh][vfat].SetPoint(dict_RawADCvsDAC_Graphs[oh][vfat].GetN(),event.dacValY,event.dacValX)
+            dict_DACvsADC_Graphs[oh][vfat].SetPointError(dict_DACvsADC_Graphs[oh][vfat].GetN()-1,calibrated_ADC_error,0)
+            dict_RawADCvsDAC_Graphs[oh][vfat].SetPointError(dict_RawADCvsDAC_Graphs[oh][vfat].GetN()-1,event.dacValY_Err,0)
+
+    # Fit the TGraphErrors objects    
+    for oh in ohArray:
+        for vfat in range(0,24):
+            #use a fifth degree polynomial to do the fit
+            dict_DACvsADC_Funcs[oh][vfat] = r.TF1("DAC Scan Function","[0]*x^5+[1]*x^4+[2]*x^3+[3]*x^2+[4]*x+[5]")
+            dict_DACvsADC_Graphs[oh][vfat].Fit(dict_DACvsADC_Funcs[oh][vfat],"Q") 
+
+    # Determine DAC values to achieve recommended bias voltage and current settings
+    graph_dacVals = {}
+    dict_dacVals = nesteddict()
+    for oh in ohArray:
+        graph_dacVals[oh] = r.TGraph()
+        for vfat in range(0,24):
+
+            maxDacValue = 255
+            
+            for dacSelect in maxVfat3DACSize.keys():
+
+                #the registers CFG_THR_ARM_DAC and CFG_THR_ZCC_DAC could correspond to voltages or currents
+                #we will use voltages until a way of distinguishing the two cases is implemented 
+                if dacSelect == 14 or dacSelect == 15:
+                    continue
+                
+                if maxVfat3DACSize[dacSelect][1] == nameX:
+                    maxDacValue = int(maxVfat3DACSize[dacSelect][0])
```

Comment:
there should be a `break` after this; e.g. once this is found why continue to loop?

Reply:
I added a break